kf_release_maker/release_maker.py

```diff
@@ -1,276 +1,198 @@
-        if version_type == 'major':
-            new_version = f'{parts[0]+1}.0.0'
-        elif version_type == 'minor':
-            new_version = f'{parts[0]}.{parts[1]+1}.0'
-        else:
-            new_version = f'{parts[0]}.{parts[1]}.{parts[2]+1}'
-
-        print(f'Next release version is {new_version}')
-
         return new_version
 
-    def create_release_markdown(self, pr_df):
-        """
-        Create release note markdown from PR DataFrame and
-        Write to a markdown file named: `<repo name>-<release version>.md`
-        """
-        print('Creating release note markdown ...\n')
-
-        # Emoji count summary markdown string
-        emoji_counts = ' '.join([
-            f'{idx} x{row["number"]}'
-            for idx, row in pr_df.groupby('emoji').count().iterrows()
-        ])
-        # Category count summary markdown string
-        endpoint = f'{self.base_url}/labels'
-        category_counts = ' '.join([
-            f'[{cat}]({endpoint}/{cat}) x{row["number"]}'
-            for cat, row in pr_df.groupby('category').count().iterrows()
-        ])
-        # PR summary markdown string
-        prs = '\n'.join(
-            [f'- (#{row["number"]}) {row["title"]} - @{row["user"]}'
-             for _, row in pr_df.iterrows()]
-        )
-
-        # Construct release notes markdown
-        title = (
-            'Kids First' + ' '.join(self.repo.split('-')).title().lstrip('Kf')
-        )
-        notes = f"# {title} Release {self.release_version}"
-        notes += '\n\n## Features\n\n'
-        notes += '### Summary\n\n'
-        notes += f'Feature Emojis: {emoji_counts} \n'
-        notes += f'Feature Categories: {category_counts} \n\n'
-        notes += '### New features and changes\n\n'
-        notes += prs
-
-        print('#'*80)
-        print(notes+'\n\n')
-
-        # Write markdown file
-        file_name = f'{self.repo}-{self.release_version}.md'
-        with open(file_name, 'w') as f:
-            f.write(notes)
+    def _to_markdown(self, repo, version, counts, prs):
+        messages = [
+            f"Release {version}",
+            "",
+            "### Summary",
+            "",
+            "- Emojis: "
+            + ", ".join(f"{k} x{v}" for k, v in counts["emojis"].items()),
+            "- Categories: "
+            + ", ".join(f"{k} x{v}" for k, v in counts["categories"].items()),
+            "",
+            "### New features and changes",
+            "",
+        ]
+
+        for p in prs:
+            userlink = f"[{p['user']['login']}]({p['user']['html_url']})"
+            sha_link = f"[{p['merge_commit_sha'][:8]}](https://github.com/{repo}/commit/{p['merge_commit_sha']})"
+            pr_link = f"[#{p['number']}]({p['html_url']})"
+            messages.append(
+                f"   {pr_link} - {p['title']} - {sha_link} by {userlink}"
+            )
 
-        print(f'Saved release notes to {file_name}')
+        return "\n".join(messages)
```

Comment:
The markdown for the commits is displaying as a paragraph instead of newline delimited strings. You might have to add another newline?

Reply:
So you want 
<img width="606" alt="Screen Shot 2020-05-13 at 2 51 42 PM" src="https://user-images.githubusercontent.com/201996/81852621-51651c00-9529-11ea-88cb-4e008ca94028.png">
instead of 
<img width="563" alt="Screen Shot 2020-05-13 at 2 51 52 PM" src="https://user-images.githubusercontent.com/201996/81852636-54f8a300-9529-11ea-93ee-7f17949c2ea0.png">
?

Reply:
what about 
<img width="572" alt="Screen Shot 2020-05-13 at 2 56 23 PM" src="https://user-images.githubusercontent.com/201996/81853047-eff17d00-9529-11ea-9721-2ab42c6b6e0e.png">

Reply:
Oh that's weird. My markdown wasn't displaying correctly. It just text wrapped all of the feature lines as if there were no new lines between each feature line.

I like the bullets though